File: selenium/java/src/test/java/demo/OtpSmsTest.java
/**
 * IMPORTANT: This test uses Mailosaur SMS, which may not
 * be enabled on your account yet (see https://mailosaur.com/app/sms)
 */

package demo;

import org.junit.Ignore;
import org.junit.Test;
import io.github.cdimascio.dotenv.Dotenv;
import com.mailosaur.MailosaurClient;
import com.mailosaur.MailosaurException;
import com.mailosaur.models.*;
import java.io.IOException;

public class OtpSmsTest {
  @Ignore("Reason")
  @Test public void retrieveOneTimePasscode() throws IOException, MailosaurException {
    Dotenv dotenv = Dotenv.load();

    String apiKey = dotenv.get("mailosaurApiKey");
    String serverId = dotenv.get("mailosaurServerId"); 

    // Add your mailosaur servers number to this variable
    String phoneNumber = dotenv.get("mailosaurPhoneNumber");   

    // Instantiate Mailosaur client with api key
    MailosaurClient mailosaur = new MailosaurClient(apiKey);

    // 1 - Perform an action that sends an otp SMS message to your number
    // https://mailosaur.com/docs/sms-testing
    // ...
    // ...

    // 2 - Create the search criteria for the sms
    // https://mailosaur.com/docs/api/messages#4-search-for-messages
    MessageSearchParams params = new MessageSearchParams();
    params.withServer(serverId).withTimeout(20000); // 20 seconds (in milliseconds)

    SearchCriteria searchCriteria = new SearchCriteria();
    searchCriteria.withSentTo(phoneNumber);

    // 3 - Get the sms from Mailosaur using the search criteria
    Message sms = mailosaur.messages().get(params, searchCriteria);

    // 4 - Retrieve passcode from sms
    // https://mailosaur.com/docs/test-cases/codes
    Code passcode = sms.text().codes().get(0);

    System.out.printf("\nSms otp code -" + passcode.value()); // "564214"
  }
}

File: go/otp_sms_test.go
/**
 * IMPORTANT: This test uses Mailosaur SMS, which may not
 * be enabled on your account yet (see https://mailosaur.com/app/sms)
 */

package mailosaurExample

import (
	"fmt"
	"os"
	"testing"

	"github.com/joho/godotenv"
	"github.com/mailosaur/mailosaur-go"
)

func TestOtpSms(t *testing.T) {
  t.Skip("reason")

  godotenv.Load()
  
  apiKey := os.Getenv("MAILOSAUR_API_KEY")
  serverId := os.Getenv("MAILOSAUR_SERVER_ID")

  // Add your mailosaur servers number to this variable
  phoneNumber := os.Getenv("MAILOSAUR_PHONE_NUMBER")

  // Instantiate Mailosaur client with api key
  m := mailosaur.New(apiKey)

  // 1 - Perform an action that sends an otp SMS message to your number
  // https://mailosaur.com/docs/sms-testing
  // ...
  // ...

  // 2 - Create the search criteria for the sms
  // https://mailosaur.com/docs/api/messages#4-search-for-messages
  params := &mailosaur.MessageSearchParams {
    Server: serverId,
    Timeout: 20000, // 20 seconds (in milliseconds)
  }

  criteria := &mailosaur.SearchCriteria {
    SentTo: phoneNumber,
  }

  // 3 - Get the sms from Mailosaur using the search criteria
  sms, err := m.Messages.Get(params, criteria)

  if (err != nil) {
    t.Error(err)
  }

  // 4 - Retrieve passcode from sms
  // https://mailosaur.com/docs/test-cases/codes
  passcode := sms.Text.Codes[0]

  fmt.Printf("\nSms otp code - " + passcode.Value + "\n")
}